src/client/datascience/jupyter/kernels/helpers.ts

```diff
@@ -63,7 +63,7 @@ export function kernelConnectionMetadataHasKernelModel(
 export function getKernelId(spec: IJupyterKernelSpec, interpreter?: PythonEnvironment) {
     // Non-Python kernels cannot contain an interpreter (even in their id).
     interpreter = isPythonKernelSpec(spec) ? interpreter : undefined;
-    return `${spec.id || ''}.${spec.name}.${spec.path}.${interpreter?.path || ''}.${
+    return `${spec.id || ''}.${spec.name}.${spec.interpreterPath || spec.path}.${interpreter?.path || ''}.${
```

Comment:
This seemed like the best way to capture the actual interpreterPath in the spec as opposed to the spec.path as one might replace the other in some scenarios like registerKernel. However if we don't have interpreterPath keep spec.path in the ID as things should not be replaced there and spec.path might be needed to distinguish ID.